### ISSUE TITLE
Handle updated streams (`stream op: update` events)

### ISSUE THREAD

**Issue by chrisbobbe** [zulip/zulip-flutter#182]
Part of #135.

https://zulip.com/api/get-events#stream-update

Handling created/deleted streams (`stream op: create` / `stream op: delete`) is #181.

I expect this will involve updating both our `store.streams` and `store.subscriptions` data structures.

**Comment by chrisbobbe** [comment]
I'm not sure how best to represent the type of this event; see in the doc, for example, `value: integer | boolean | string`.

**Comment by chrisbobbe** [comment]
Ah: removing the "beta" milestone until we make that choice explicitly for this event. From Greg's comment on the umbrella issue https://github.com/zulip/zulip-flutter/issues/135#issuecomment-1565081678 :

> I've given this the "beta" milestone, but there may be specific events we decide can wait for launch or even beyond. What's needed for beta is really just that we should look at the list and make any such choice explicitly.

**Comment by gnprice** [comment]
> I'm not sure how best to represent the type of this event; see in the doc, for example, `value: integer | boolean | string`.

Update: a couple of months after this comment, we added `UserSettingsUpdateEvent`, which has a very similar pattern in the API:
https://zulip.com/api/get-events#user_settings-update
and we worked out a pattern for handling that which we're pretty happy with. So for adding `StreamUpdateEvent`, let's follow that pattern.

To see all the different pieces of that pattern, see commits 9cb2e4d0f6300693186c9231a570e03050bd6d18 and 4bcbffd529663737c6e3480dfd477bfcd70aa7c8 (both from #261). I think a good PR for this issue will look a lot like those two commits.
